azext_iot/tests/iothub/devices/test_hub_export_import.py

```diff
@@ -0,0 +1,364 @@
+import random
+import json
+import os
+import time
+from azext_iot.common.shared import DeviceAuthApiType
+
+from azext_iot.tests.settings import DynamoSettings, ENV_SET_TEST_IOTHUB_REQUIRED, ENV_SET_TEST_IOTHUB_OPTIONAL
+from azext_iot.tests.iothub import IoTLiveScenarioTest
+from azext_iot.tests.generators import generate_generic_id
+from azext_iot.common.utility import generate_key
+from azext_iot.common.certops import create_self_signed_certificate
+from azext_iot.tests.iothub import (
+    PRIMARY_THUMBPRINT,
+    SECONDARY_THUMBPRINT,
+    DEVICE_TYPES,
+)
+
+settings = DynamoSettings(req_env_set=ENV_SET_TEST_IOTHUB_REQUIRED, opt_env_set=ENV_SET_TEST_IOTHUB_OPTIONAL)
+
+
+class TestHubExportImport(IoTLiveScenarioTest):
+    def __init__(self, test_case):
+        super(TestHubExportImport, self).__init__(test_case)
+        self.dest_hub = settings.env.azext_iot_desthub or "test-hub-" + generate_generic_id()
+        self.dest_hub_rg = settings.env.azext_iot_destrg or settings.env.azext_iot_testrg
+
+        # create destination hub
+
+        if not settings.env.azext_iot_desthub:
+            self.create_hub(self.dest_hub, self.entity_rg)
+
+        self.dest_hub_cstring = self.cmd(
+            "iot hub connection-string show -n {} -g {}".format(self.dest_hub, self.dest_hub_rg)
+        ).get_output_in_json()["connectionString"]
+
+        self.filename = generate_generic_id() + ".json"
+
+        self.create_hub_state()
+
+    def create_hub_state(self):
+
+        # make a configuration for the hub (applies to 0 devices, this is just to test the configuration settings)
+
+        config_content = {"deviceContent" : {
+            "properties.desired.propFromConfig" : generate_generic_id()
+        }}
+        config_content = json.dumps(config_content)
+
+        labels = {generate_generic_id() : generate_generic_id(), generate_generic_id() : generate_generic_id()}
+        labels = json.dumps(labels)
+
+        priority = random.randint(1, 10)
+
+        metrics = {"metrics": {
+            "queries": {
+                "mymetric": "SELECT deviceId from devices where properties.reported.lastDesiredStatus.code = 200"
+            }
+        }}
+        metrics = json.dumps(metrics)
+
+        self.kwargs["config_content"] = config_content
+        self.kwargs["labels"] = labels
+        self.kwargs["metrics"] = metrics
+
+        self.cmd("iot hub configuration create --config-id {} -l {} --content '{}' --labels '{}' --priority {} --metrics '{}' \
```

Comment:
ah this tabbing is weird - stay consistent (like the tabbing/spacing for metrics)